script/bulk_import/generic_bulk.rb

```diff
@@ -2006,6 +2007,66 @@ def import_votes
     puts "  Update took #{(Time.now - start_time).to_i} seconds."
   end
 
+  def import_topic_voting_votes
+    unless defined?(::DiscourseTopicVoting)
+      puts "", "Skipping topic voting votes, because the topic voting plugin is not installed."
+      return
+    end
+
+    puts "", "Importing votes for topics..."
+
+    topic_votes = query(<<~SQL)
+      SELECT *
+      FROM topic_voting_votes
+    SQL
+
+    existing_topic_votes = DiscourseTopicVoting::Vote.pluck(:topic_id, :user_id).to_set
+
+    create_topic_voting_votes(topic_votes) do |row|
+      topic_id = topic_id_from_imported_id(row["topic_id"])
+      user_id = user_id_from_imported_id(row["user_id"])
+
+      next unless topic_id && user_id
+      next unless existing_topic_votes.add?([topic_id, user_id])
+
+      {
+        topic_id: topic_id,
+        user_id: user_id,
+        archive: to_boolean(row["archive"]),
+        created_at: to_datetime(row["created_at"]),
+        updated_at: to_datetime(row["updated_at"]),
+      }
+    end
+
+    topic_votes.close
+
+    puts "", "Updating vote counts of topics..."
+
+    start_time = Time.now
+
+    DB.exec(<<~SQL)
+      WITH missing_ids AS (
+        SELECT DISTINCT t.id FROM topics t
+        JOIN topic_voting_votes dvv ON t.id = dvv.topic_id
```

Comment:
Is the `JOIN` on the `topics` table really needed?

Reply:
I left the queries as they are from the Topic Votes Plugin [20240711102255_ensure_consistency.rb](https://github.com/discourse/discourse-topic-voting/blob/a39bb09bc9d17d1dc77084f20eb0ddd77c97b9a9/db/post_migrate/20240711102255_ensure_consistency.rb#L43-L63) file.